### PR TITLE
go-github update

### PR DESCRIPTION
**Pull request by andrew-waters**
https://github.com/google/go-github/releases/tag/v32.0.0 has been released and included breaking changes to a few functions in this provider. This PR aims to maintain parity with existing functionality but will unblock work on organisation secrets (https://github.com/terraform-providers/terraform-provider-github/issues/468)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,7 @@ require (
 	github.com/client9/misspell v0.3.4
 	github.com/golangci/golangci-lint v1.25.1
 	github.com/google/go-github/v31 v31.0.0
```
[the remark this answers is not in the document]
Comment (andrew-waters) [reply go.mod]: @anGie44 apologies for the delay getting  back to you here - I've just committed 79d396e6c0472e69c9ec168b110905dae13d5b2a to get rid of v31 completely.

The underlying changes you've created an issue for I'm also subscribed to so let's see what comes of that conversation - it doesn't appear to have been resolved.

[the remark this answers is not in the document]
Comment (anGie44) [reply go.mod]: no worries @andrew-waters, thanks for having a look! reminds me I should check-in on that issue

**Comment by ejhayes** [comment]
It looks like the tests are currently passing with these changes. Is this something that can be merged in or are you still waiting on https://github.com/google/go-github/issues/1546?

**Comment by hitmands** [comment]
Is there any update on this regards?

**Comment by andrew-waters** [comment]
I'd refer to @anGie44 - as she ran the tests that picked up the issue, it would be good to see if they now pass as is suggested

**Comment by anGie44** [comment]
hi @hitmands, @andrew-waters! Unfortunately without upstream changes in the go-github library, the acceptance tests discussed above are still failing (perhaps you were referring to the Travis CI checks  @ejhayes?) good news is the PR to revert that breakage is in review https://github.com/google/go-github/pull/1571 and once that's in we can pick this one back up 👍

**Comment by andrew-waters** [comment]
@anGie44 as you're already aware https://github.com/google/go-github/pull/1571 has now been merged. Let me know if this PR needs any further work

**Comment by andrew-waters** [comment]
I've upgraded the lib to the newly released [v32.1.0](https://github.com/google/go-github/releases/tag/v32.1.0).

I'd be very grateful if someone can run the acceptance tests to ensure that the release has fixed the issue we faced.

**Comment by ejhayes** [comment]
@andrew-waters looks like this can be merged in any time.

**Comment by andrew-waters** [comment]
Can I self merge @ejhayes ? First contribution :)

**Comment by ejhayes** [comment]
Good point @andrew-waters. @anGie44 are you able to merge this change in?

**Comment by ejhayes** [comment]
@hitmands are you able to merge this PR?

**Comment by nikolay** [comment]
@hitmands Any chance to merge this?

**Comment by ejhayes** [comment]
@jcudit are you able to merge this PR?

**Comment by jcudit** [comment]
@ejhayes yes, but in time.  This change and the feature it unblocks is tracked to be released at the end of September.  Focus is currently on cleaning up testing and releasing [v3.0.0](https://github.com/terraform-providers/terraform-provider-github/milestone/12).  Once that blocker is cleared, we can get back to shipping changes like this at a more frequent pace.  🙇 for the patience on this one.

**Review by hitmands — APPROVED** [review]

**Comment by jcudit** [comment]
Resolving conflicts seems to have broken the tests in some way 🙃 .  I've merged b587cb0 to accomplish the upgrade instead.

**Comment by andrew-waters** [comment]
Thanks for adding this feature @jcudit. I'm glad that the proposed functionality can be achieved now as it is functionality I can leverage.

I'm somewhat disappointed that this PR had to be dropped - It seems that a lack of urgency (the PR was first cut on May 28th) turned into an abundance of urgency (the merge took less than 24 hours with a fresh cut) and I didn't get the opportunity to resolve them myself.

**Comment by jcudit** [comment]
Apologies @andrew-waters, fair concern.

I have been working through a transition with primary responsibilities. Time has been sparse for this project and I've been more aggressive with changes to compensate.

I expect us to get a few more maintainers and return to a more predictable pace in the coming months.

**Pull request closed without merging**